### DISCUSSION TITLE
How to prevent brain crack

### DISCUSSION THREAD

**Discussion by wangDong524** [crate/crate#11640] [Q&A]
Hi,
 I see the setting of "discovery.zen.minimum_master_nodes" is removed  in the crateDB version 4.0.0. This version of CrateDB uses a new cluster coordination (discovery) implementation which improves resiliency and master election times. A new voting mechanism is used when a node is removed or added which makes the system capable of automatically maintaining an optimal level of fault tolerance even in situations of network partitions.
 I whant to know  How to prevent brain crack  ,or the way to prevent brain cracking？
Thanks

**Comment by jayeff** [comment]
Hi,
the new voting system with 4.0.0 improves resiliency which includes network partitions. You don't need to worry specifically about this.

Here is some background reading on this topic: https://www.elastic.co/blog/a-new-era-for-cluster-coordination-in-elasticsearch

**Reply by wangDong524** [reply]
thanks.
I have another question:
How to get the version  of elasticsearch corresponding to cratedb,example crateDB 4.6.1

**Reply by seut** [reply]
Since CrateDB >= 4.0.0, we do not use an upstream version of the Elasticsearch repository anymore. We've integrated the parts we are using from it (mainly discovery, coordination, replication, routing, sharding) into our code-base. Since then we backport relevant commits from the elasticsearch master branch. Thus there is no exact direct version relationship between CrateDB releases and Elasticsearch releases since then but very roughly speaking its currently somewhere like CrateDB major-version + 3 = ES major version (CrateDB 4.x ~= ES 7.x).

Due to recent license changes at the Elasticsearch code, this may change even more soon as we're forced to switch to a fork of ES (e.g. OpenSearch) to get further backport into our code.

**Reply by wangDong524** [reply]
Oh, I see.Thank you for your reply
